src/main/java/cs2103/v15_1j/jimjim/ui/TaskEventRowFactory.java

```diff
@@ -294,7 +307,13 @@ else if(!event.getCompleted()){
 	}
 
 	private void addTaskEvent(DeadlineTask task){
-		displayList.addWithoutSorting(task);
+		int id = displayList.size('d') + 1;
+		if(displayList.contains(task)){
```

Comment:
I think it's better to just check whether id == 0 instead of iterating through the list again.